component/block_cache/block_cache.go

```diff
@@ -350,18 +350,17 @@ func (bc *BlockCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Han
 			return nil, fmt.Errorf("failed to retrieve block list for %s", options.Name)
 		}
 
+		valid := bc.validateBlockList(options.Name, bc.blockSize, blockList)
```

Comment:
blockSize is a class member only no need to pass it down to another method.
Ideally we shall just send the handle and the blockList and blockList map updation shall be done in the new method.